src/peft/tuners/lora/config.py

```diff
@@ -69,6 +69,31 @@ class LoftQConfig:
     loftq_iter: int = field(default=1, metadata={"help": "Alternating iterations for LoftQ"})
 
 
+@dataclass
+class EvaConfig:
+    """
+    This is the sub-configuration class to store the configuration for a data-drive initialization via EVA.
+
+    Args:
+        rho (`float`): Rho value for EVA redistribution. Default is 1.0.
+        tau (`float`): Cosine similarity threshold for early stopping. Default is 0.99.
+        use_label_mask (`bool`): Use label mask for EVA initialization. Default is False.
+        label_mask_value (`int`): If use_label_mask=True the value to look for to mask out ignored tokens. Default is -100.
+        whiten (`bool`): Apply whitening to singular vectors. Default is False.
+        device (`str`): Device to use for EVA initialization. Default is 'cuda'.
+    """
+
+    rho: float = field(default=1.0, metadata={"help": "Rho value for EVA redistribution"})
+    tau: float = field(default=0.99, metadata={"help": "Cosine similarity threshold for early stopping"})
+    use_label_mask: bool = field(default=False, metadata={"help": "Use label mask for EVA initialization"})
+    label_mask_value: int = field(default=-100, metadata={"help": "if use_label_mask=True the value to look for to mask out ignored tokens"})
+    whiten: bool = field(default=False, metadata={"help": "Apply whitening to singular vectors"})
+    device: str = field(default="cuda", metadata={"help": "Device to use for EVA initialization"})
+
+    def __post_init__(self):
+        assert self.rho >= 1.0, "early_stop_rho must be >= 1"
```

Comment:
Let's completely avoid all `assert`s except for tests. Instead, raise a `ValueError` here. Same for the other asserts used below.